dynawo/sources/Solvers/SolverIDA/DYNSolverIDA.cpp

```diff
@@ -732,7 +732,7 @@ SolverIDA::reinit() {
   if (modeChangeType == NO_MODE) return;
 
   const bool evaluateOnlyMode = optimizeReinitAlgebraicResidualsEvaluations_;
-  if (modeChangeType != DIFFERENTIAL_MODE) {
+  if (modeChangeType >= minimumModeChangeTypeForAlgebraicRestoration_) {
```

Comment:
You can report the change here regarding NO_MODE test

Reply:
sorry I missed that it was also in IDA. Let me update

Reply:
This might not be true in this case as there is a call to IDAReInit that should not be made when there is no mode change if I'm not mistaken

Reply:
Yes, you are totally right.
I completely forgot this